src/ert/_c_wrappers/enkf/config/ext_param_config.py

```diff
@@ -95,36 +91,51 @@ def __getitem__(self, index):
         that index
         An IndexError is raised if the item is not found
         """
+        suffixes = []
+
         if isinstance(index, str):
-            i = self._key_index(index)
-            if i < 0:
-                raise IndexError(f'Key "{index}" not found')
-            return self._get_suffixes(i)
+            suffix_index = self._get_key_index(index)
+            if self._suffix_list_contains_index(suffix_index):
+                suffixes = self._key_suffix_value(suffix_index)
+            return suffixes
 
         # assume index is an integer
-        if index < 0:
-            return index + len(self)
-        if index >= len(self):
+        if self._suffix_list_contains_index(index):
+            suffixes = self._key_suffix_value(index)
+        key = self._key_list[index]
+
+        return key, suffixes
+
+    def _get_key_index(self, key: str) -> int:
+        if self._key_list.count(key) == 0:
             raise IndexError(
-                f"Invalid key index {index}. Valid range is [0, {len(self)})"
+                f"Requested index not found: {key}, " f"Keylist: {self._key_list}"
             )
-        key = self._iget_key(index)
-        suffixes = self._get_suffixes(index)
-        return key, suffixes
+        return self._key_list.index(key)
```

Comment:
In know the old code did it this why but what do you think of updating this logic to something like:

```python
try:
     return self._key_list.index(key)
except ValueError:
     raise IndexError(
          f"Requested index not found: {key}, " f"Keylist: {self._key_list}"
     ) 
```